packages/docusaurus-utils/src/tags.ts

```diff
@@ -7,25 +7,45 @@
 
 import {kebabCase, uniq, uniqBy} from 'lodash';
 import {normalizeUrl} from './urlUtils';
+import logger from '@docusaurus/logger';
 
 export type Tag = {
   label: string;
   permalink: string;
 };
 
-export type FrontMatterTag = string | Tag;
+export type FrontMatterTag = string | Partial<Tag>;
+export type TagsMap = Record<string, Partial<Tag>>;
```

Comment:
The problem is that we may add more stuff in the future like description, permalink, etc. I feel like it's valuable to make it remain partial, because maybe not everything makes sense to have a default?

Reply:
@Josh-Cena I don't imagine any case where a tag without label make sense.

At the end of the day, we must display that tag somewhere in the UI, so there should always be at least a label